sphinx/themes/basic/static/searchtools.js

```diff
@@ -293,10 +293,12 @@ const Search = {
       if (title.toLowerCase().trim().includes(queryLower) && (queryLower.length >= title.length/2)) {
         for (const [file, id] of foundTitles) {
           let score = Math.round(100 * queryLower.length / title.length)
+          let isDocumentTitle = titles[file] === title
+          let sectionId = id ? "#" + id : ""
           results.push([
             docNames[file],
-            titles[file] !== title ? `${titles[file]} > ${title}` : title,
-            id !== null ? "#" + id : "",
+            isDocumentTitle ? title : `${titles[file]} > ${title}`,
+            isDocumentTitle ? "" : sectionId, // don't use the section id if we matched on the exact document title (creates duplicates below)
```

Comment:
Is there _always_ an entry in `foundTitles` that is the title of the document?  (and does it have a special/fixed `id` value if so?)

Reply:
Ok.  Basically I'm wondering whether all of the `title` elements from `foundTitles` are already in the `foundEntries` index -- and if so, whether we could omit the [`foundTitles` loop](https://github.com/sphinx-doc/sphinx/blob/27f4d3605989ea2216c5f370b07833a501de7267/sphinx/themes/basic/static/searchtools.js#L292-L308) entirely.

Reply:
> Ok. Basically I'm wondering whether all of the `title` elements from `foundTitles` are already in the `foundEntries` index -- and if so, whether we could omit the [`foundTitles` loop](https://github.com/sphinx-doc/sphinx/blob/27f4d3605989ea2216c5f370b07833a501de7267/sphinx/themes/basic/static/searchtools.js#L292-L308) entirely.

It seems that it is _not_ possible to omit that loop.  The only way to find nested sections (`A > B`) seems to be from results in `foundTitles` (I tested with query `quick guide` for the Sphinx self-build `doc` dir).

Reply:
Yeah index entries are their own concept, different from titles:

https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#index-generating-markup
https://coderefinery.github.io/sphinx-lesson/indexing/#indexing

Reply:
So if I understand: `titles` have a two-level hierarchy -- top-level titles, and then items found 'within' that title.  That index is created from the section headings in RST documents.

Meanwhile `entries` are flat / have no hierarchy.  That index is created from directives, options, domain entities and so on.

And in cases where a `title` and an `entry` overlap, we want to de-duplicate.

Reply:
> And in cases where a title and an entry overlap, we want to de-duplicate.

To clarify that: in cases where a _top-level_ title and an _entry without an anchor_ overlap (?).

Reply:
Hrm, something's off about my understanding here.

* :white_check_mark: I can search the Sphinx documentation for `translating with sphinx-intl`, and I see a second-level title result: `Internationalization > Translating with sphinx-intl`.
* :x: But I _cannot_ seem to search for `using weblate` -- if I scroll I eventually find a single, top-level `Internationalization` entry -- but I would have expected to find `Internationalization > Using Weblate service for team translation`.

Reply:
I'm going to take a break from reviewing for a while @wlach - I'll be back within the next few days.  Thank you for looking into this.

Reply:
> ❌ But I cannot seem to search for using weblate -- if I scroll I eventually find a single, top-level Internationalization entry result -- but I would have expected to find Internationalization > Using Weblate service for team translation.

"using weblate" is too short a search term for it to match (since the full title is 42 characters long, more than double the size of the section item):

https://github.com/wlach/sphinx/blob/8daf8c997340e67aa77db3a78a2140eaebb8a3ab/sphinx/themes/basic/static/searchtools.js#L293

If you search for e.g. "Using weblate service" you'd get the result you expected above.

> I'm going to take a break from reviewing for a while @wlach - I'll be back within the next few days. Thank you for looking into this.

No problem, thank you! I may have limited time/opportunity to revise this PR for a few weeks after this Wednesday, but I won't forget about it. Also welcome maintainer changes / updates.